pkg/azure/actuator.go

```diff
@@ -492,3 +520,102 @@ func (a *Actuator) getLogger(cr *minterv1.CredentialsRequest) log.FieldLogger {
 func (a *Actuator) Upgradeable(mode operatorv1.CloudCredentialsMode) *configv1.ClusterOperatorStatusCondition {
 	return utils.UpgradeableCheck(a.client.RootCredClient, mode, a.GetCredentialsRootSecretLocation())
 }
+
+// Generated Secret needs to look like this:
+/*
+	apiVersion: v1
+	stringData:
+	  azure_client_id: 0420bfd1-ab26-4b80-a9ac-deadbeeff1f9
+	  azure_tenant_id: 6047c7e9-b2ad-488d-a54e-deadbeefa7ee
+	  azure_region: centralus
+	  azure_subscription_id: 8c20ec23-8478-4f46-96f4-deadbeeff185
+	  azure_federated_token_file: /var/run/secrets/openshift/serviceaccount/token
+	kind: Secret
+	metadata:
+	  name: azure-cloud-credentials
+	  namespace: openshift-machine-api
+	type: Opaque
+*/
+// The first 4 fields need to come from `spec.ProviderSpec` in the CredentialsRequest with
+// spec.cloudTokenPath matching up to: `azure_federated_token_file`
+func (a *Actuator) syncWorkloadIdentitySecret(
+	azureProviderSpec minterv1.AzureProviderSpec,
+	azureFederatedTokenFile string,
+	cr *minterv1.CredentialsRequest,
+	logger log.FieldLogger, ctx context.Context) error {
+	sLog := logger.WithFields(log.Fields{
+		"targetSecret": fmt.Sprintf("%s/%s", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name),
+		"cr":           fmt.Sprintf("%s/%s", cr.Namespace, cr.Name),
+	})
+	sLog.Infof("processing secret")
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Spec.SecretRef.Name,
+			Namespace: cr.Spec.SecretRef.Namespace,
+		},
+	}
+	err := validateAzureProviderSpec(azureProviderSpec)
+	if err != nil {
+		logger.WithError(err).Error("error determining whether a credentials update is needed")
+		msg := "error validating credentials request Azure AD Workload Identity fields"
```

Comment:
maybe we don't need this message? the error coming from the validate method already includes everything. Wdyt?